src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -10,18 +10,91 @@
  */
 
 #include "vulnerabilityScannerFacade.hpp"
+#include "archiveHelper.hpp"
 #include "flatbuffers/include/syscollector_deltas_generated.h"
 #include "flatbuffers/include/syscollector_synchronization_generated.h"
 #include "loggerHelper.h"
+#include "timeHelper.h"
 #include "vulnerabilityScanner.hpp"
 #include "wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp"
+#include "xzHelper.hpp"
 
 constexpr auto VULNERABILITY_SCANNER_TEMPLATE = "queue/indexer/vd_states_template.json";
 constexpr auto DEFAULT_QUEUE_PATH = "queue/sockets/queue";
 constexpr auto REPORTS_QUEUE_PATH = "queue/vd/reports";
 constexpr auto REPORTS_BULK_SIZE {1};
 constexpr auto REPORTS_THREADS_NUMBER {1};
 constexpr auto MICROSEC_FACTOR {1000000};
+constexpr auto VD_STATE_QUEUE_PATH = "queue/vd/state_track";
+constexpr auto VD_KEYSTORE_PATH = "queue/keystore";
+constexpr auto VD_DATABASE_PATH {"queue/vd"};
+
+bool VulnerabilityScannerFacade::decompressDatabase(const std::string& databaseVersion)
+{
+    bool ret = false;
+
+    // Check database version and presence. It will attempt to decompress the database
+    // if the version doesn't not match, the state_track does not have information
+    // or there's no VD DB.
+    if (databaseVersion.compare(__ossec_version) != 0 || databaseVersion.empty() ||
+        !std::filesystem::exists(VD_DATABASE_PATH))
+    {
+        // Search in path for a compressed database file.
+        std::regex regexPattern(VD_COMPRESSED_FILE_PATTERN);
```

Comment:
Question (design): What if there are multiple compressed files? Couldn't we make the check based on the value of __ossec_version?

Reply:
I didn't want to add too much logic for this case, I commented on the issue about this case, I thought about it but decided to keep it simple, let me know if it is a wrong approach.

Reply:
Changes according to the discussion we had